### PR TITLE
8343774: Positive list platforms for ir checks of compiler/c2/TestCastX2NotProcessedIGVN.java

### PR DESCRIPTION
**Pull request by reinrich**
Vectorization of the loop in `test2` does not work on ppc therefore I want to exclude it there.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343774](https://bugs.openjdk.org/browse/JDK-8343774): Positive list platforms for ir checks of compiler/c2/TestCastX2NotProcessedIGVN.java (**Sub-task** - P4)

### Reviewers
 * [Roland Westrelin](https://openjdk.org/census#roland) (@rwestrel - **Reviewer**) Review applies to [a7c2872b](https://git.openjdk.org/jdk/pull/21975/files/a7c2872b86ec6b43562cdc49403347cba16bdbc4)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Amit Kumar](https://openjdk.org/census#amitkumar) (@offamitkumar - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21975/head:pull/21975` \
`$ git checkout pull/21975`

Update a local copy of the PR: \
`$ git checkout pull/21975` \
`$ git pull https://git.openjdk.org/jdk.git pull/21975/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21975`

View PR using the GUI difftool: \
`$ git pr show -t 21975`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21975.diff">https://git.openjdk.org/jdk/pull/21975.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21975#issuecomment-2464893298)
</details>

### CODE REVIEW

**Comment by reinrich** [comment]
@offamitkumar want me to exclude s390x as well? @MBaesken said TestCastX2NotProcessedIGVN.java was failing there too.

**Comment by reinrich** [comment]
> Thanks for taking care of that. Maybe it would be more robust to only enable the test on x86, aarch64 and riscv64. The whole test doesn't need to excluded actually. Only IR matching on `test2` needs to be disabled. This can be done with:
> 
> ```
> applyIfPlatformOr = {"x64", "true", "aarch64", "true", "riscv64", "true"})
> ```
> 
> See `TestBoolNodeGVN.java` for instance.

Ok. I've done that.

**Review by rwestrel — APPROVED** [review]
Looks good to me.

File: test/hotspot/jtreg/compiler/c2/TestCastX2NotProcessedIGVN.java [hunk 1]
```diff
@@ -62,7 +62,8 @@ public static void test1Runner() {
     }
 
     @Test
-    @IR(counts = {IRNode.LOAD_VECTOR_I, "> 1"})
+    @IR(counts = {IRNode.LOAD_VECTOR_I, "> 1"},
+        applyIfPlatformOr = {"x64", "true", "aarch64", "true", "riscv64", "true"})
```
Comment (RealFYang) [inline test/hotspot/jtreg/compiler/c2/TestCastX2NotProcessedIGVN.java:66]: Hi, Could you please remove `riscv64` from this line? I just found that this test also fails when testing on riscv64 platforms where the vector extension is not available. Thanks.

Comment (reinrich) [reply test/hotspot/jtreg/compiler/c2/TestCastX2NotProcessedIGVN.java]: Sure. I've removed it.

**Review by RealFYang — APPROVED** [review]
LGTM. Thanks.

**Review by offamitkumar — APPROVED** [review]

**Comment by offamitkumar** [comment]
@reinrich Sorry for creating mess here.

Yesterday, this test failed while testing changes for JEP 450 related to compact headers. However, now I checked and head stream testing job shows that it does not fail with `jdk-head`;

 I have verified and It fails only on s390x when I enable UseCompactObjectHeaders:  `make test TEST=jtreg:$(find . -name TestCastX2NotProcessedIGVN.java) JTREG="VM_OPTIONS=-XX:+UnlockExperimentalVMOptions -XX:+UseCompactObjectHeaders"`.

While this issue could potentially occur with other settings, `UseCompactObjectHeaders` is the only one I have observed causing this failure. Do you suggest disabling this, or is separate debugging required to investigate this behaviour?"

**Comment by reinrich** [comment]
It's really up to you @offamitkumar. For PPC we have opened an internal bug (actually it should be mirrored by a JBS-issue) to revise the compilation of `test2`.

**Comment by reinrich** [comment]
I've converted this issue [JDK-8343774](https://bugs.openjdk.org/browse/JDK-8343774) into a subtask. In the subtask the platforms where the ir checks of `test2`  succeed are positive listed.

The issues of other platforms are tracked in the parent task.

**Comment by offamitkumar** [comment]
> It's really up to you @offamitkumar. For PPC we have opened an internal bug (actually it should be mirrored by a JBS-issue) to revise the compilation of `test2`.

I did same yesterday for s390x as well. I have added it in todo list for the internal tracker. Maybe If possible, you can add s390x as affected architecture for JBS issue: [JDK-8343906](https://bugs.openjdk.org/browse/JDK-8343906).

I am fine now with integrating it.

**Comment by reinrich** [comment]
> just a minor title update `Positiv` -> `Positive`

Thanks :)

**Comment by reinrich** [comment]
> > It's really up to you @offamitkumar. For PPC we have opened an internal bug (actually it should be mirrored by a JBS-issue) to revise the compilation of `test2`.
> 
> I did same yesterday for s390x as well. I have added it in todo list for the internal tracker. Maybe If possible, you can add s390x as affected architecture for JBS issue: [JDK-8343906](https://bugs.openjdk.org/browse/JDK-8343906).

I've added s390x. Please feel free to add details if needed.

> I am fine now with integrating it.

Ok, thanks.

**Pull request closed without merging**